### PR TITLE
fix: fallback for date formats in sqlite parsing

### PR DESCRIPTION
**Pull request by tychoish**
@talagluck: do you want to take a look at adding some better test
cases. Since `INSERT INTO` works for sqlite, we can do some
interesting local stuff

### CODE REVIEW

**Review by scsmithr — APPROVED** [review]

**Comment by talagluck** [comment]
@tychoish - I pushed a few tests. Are you able to point me to INSERT INTO tests for sqlite? I didn't see anything.

File: testdata/sqllogictests_sqlite/read.slt [hunk 1]
```diff
@@ -4,3 +4,26 @@ query I
...
+93
+
+# TODO: change db location to our own test directory on github once this is merged
+query I
+SELECT COUNT(*) FROM read_sqlite('https://github.com/jpwhite3/northwind-SQLite3/raw/main/dist/northwind.db','Orders');
+----
+16282
+
+query III
+SELECT OrderID, OrderDate, RequiredDate, ShippedDate FROM FROM read_sqlite('https://github.com/jpwhite3/northwind-SQLite3/raw/main/dist/northwind.db','Orders');
```
Comment (tychoish) [inline testdata/sqllogictests_sqlite/read.slt:20]: I don't know how I feel about committing a 25mb file to our repo just for these tests. 

I think you can either:
- create a very small database with a subset of this data in the repo and commit it
- upload it to the google bucket with the other (larger) test databases.
- in SLTs create a database with data in the desired forms.

File: testdata/sqllogictests_sqlite/data/setup-test-sqlite-db.sql [hunk 2]
```diff
@@ -110,5 +110,35 @@ CREATE TABLE IF NOT EXISTS bikeshare_trips (
...
+
+CREATE TABLE IF NOT EXISTS date_test(
+    datetime_a DATETIME,
+    datetime_b DATETIME,
+    text_a TEXT
+);
+
+INSERT INTO date_test (datetime_a, datetime_b, text_a) VALUES
+('2016-07-04', '13:46:17', '2012-12-26 04:58:22'),
+('2022-02-09','23:37:27', '2016-07-11' ),
```
Comment (tychoish) [inline testdata/sqllogictests_sqlite/data/setup-test-sqlite-db.sql:143]: I'm not sure what the desired semantics for "timestamp [datetime] without date part" is. 

This is presently an error, and that seems correct.

Comment (talagluck) [inline testdata/sqllogictests_sqlite/data/setup-test-sqlite-db.sql:143]:
```suggestion
('2016-07-04', '2016-07-04 13:46:17', '2012-12-26 04:58:22'),
('2022-02-09', '2016-07-04 23:37:27', '2016-07-11' ),
```

**Pull request merged**